lib/hiera/backend/vault_backend.rb

```diff
@@ -92,22 +106,29 @@ def lookup_generic(key, scope)
           end
 
           return nil if secret.nil?
-
           Hiera.debug("[hiera-vault] Read secret: #{key}")
           if @config[:default_field] and (@config[:default_field_behavior] == 'ignore' or (secret.data.has_key?(@config[:default_field].to_sym) and secret.data.length == 1))
```

Comment:
Default fields also need to handle the kv2 case (there should be a `secret.data[:data].has_key?(@config[:default_field]` case in here somewhere).